functions/New-DbaDbUser.ps1

```diff
@@ -0,0 +1,247 @@
+                    #     Write-Message -Level Verbose -Message "Using ParamaterSetName: Contained"
+                    #     # Is this database contained?
+                    #     If($db.ContainmentType -ne "Partial" ){
+                    #         Stop-Function -Message "Sql User's with passwords can only be created on partially contained databases."  -Category InvalidOperation -Target $instance;
+                    #         return
+                    #     } else {
+                    #         Write-Message -Level Verbose -Message $db.ContainmentType
+                    #     }
+                    
+                    
+                    #     $UserType = [Microsoft.SqlServer.Management.Smo.UserType]::SqlUser
+                    
+                    # }
+                    
+                    "NoLogin" {
+                        # Creates a user without login
+                        Write-Message -Level Verbose -Message "Using UserType: NoLogin"
+                        $UserType = [Microsoft.SqlServer.Management.Smo.UserType]::NoLogin
+                        $Name = $Username;
+                    }
+                } #switch
+                
+                # Does user exist with same name?
+                if ($existingUser = $db.Users[$Name]) {
+                    if ($force) {
+                        if ($Pscmdlet.ShouldProcess($existingUser, "Dropping existing user $Name because -Force was used")) {
+                            try {
+                                $existingUser.Drop()
+                            }
+                            catch {
+                                Stop-Function -Message "Could not remove existing user $Name, skipping." -Target $Name -Continue
+                                return
+                            }
+                        }
+                    }
+                    else {
+                        Stop-Function -Message "User $Name already exists and -Force was not specified" -Target $Name
+                        return
+                    }
+                }
+                
+                
+                if ($Pscmdlet.ShouldProcess($db, "Creating user $Name")) {
+                    try {
+                        $smoUser = New-Object Microsoft.SqlServer.Management.Smo.User;
+                        $smoUser.Parent = $db;
+                        $smoUser.Name = $Name;
+                        
+                        If ($PSBoundParameters.Keys -contains 'Login' -and $Login.GetType().Name -eq 'Login') { $smoUser.Login = Login }
+                        $smoUser.UserType = $UserType;
+                        
+                        $smoUser.Create();
+                        
+                    }
+                    catch {
+                        Stop-Function -Message "Failed to add user $Name in $db to $instance" -Category InvalidOperation -ErrorRecord $_ -Target $instance -Continue;
+                        return
```

Comment:
This is a contradiction, to have continue on `Stop-Function` but also include the `return`. Need to remove return if you want the command to continue processing after the catch block is hit.